content/general/concepts/jit-compilation/jit-compilation.md

```diff
@@ -0,0 +1,41 @@
+  - 'introduction-to-javascript'
+  - 'paths/computer-science'
+---
+
+Just-In-Time (JIT) compilation is a system for compiling languages as they are executed, rather than compiling prior to execution. In JIT compilation, the compiling happens during the [runtime](https://www.codecademy.com/resources/docs/general/runtime), which is a built-in application that supports the execution of a program. Runtime generally refers to the infrastructure required to implement features in the language itself.
+
+JIT compilation can be thought of as a combination of traditionally compiled languages and interpreted languages. Traditionally compiled languages are languages where the computer directly translates the code and program prior to execution. Interpreted languages are not translated by the computer, but instead, they use an interpreter to read and execute the code directly. [C](https://www.codecademy.com/resources/docs/c) is an example of a traditionally compiled language, and [Python](https://www.codecademy.com/resources/docs/python) is an example of an interpreted language. JIT compilation is a mixture of these two methods, and it directly compiles into [machine code](https://www.codecademy.com/resources/docs/general/machine-code) at runtime.
+
+## Use Cases for JIT compilation
+
+Some major implementations of JIT compilation include the Java Virtual Machine (JVM) and the Common Language Runtime (CLR). JVM is used for [Java](https://www.codecademy.com/resources/docs/java) and Kotlin, and CLR is used for C#. [JavaScript](https://www.codecademy.com/resources/docs/javascript) also uses JIT compilation, but the specific implementation depends on the browser. Low-level languages, such as [C](https://www.codecademy.com/resources/docs/c) or [C++](https://www.codecademy.com/resources/docs/cpp), do not use JIT compilation, because low-level languages do not use a runtime.
+
+## Benefits of JIT compilation
+
+There are multiple benefits of JIT compilation, such as:
+
+### Improved Performance for Interpreted Languages
+
+Compiled languages are more performant than interpreted languages. JIT compilation allows interpreted languages to have the increased performance benefits of compilation, without the overhead that comes with traditional compiling.
+
+### Increased Efficiency for Compiled Languages
+
+With traditionally compiled languages, programmers must wait for their code to compile prior to starting the execution of a program. JIT compilation does not include this wait time, since JIT compilation compiles the code during the execution phase. Removing this extra step increases efficiency for programmers and the development process.
+
+### Decreased Execution Time
+
+JIT compilation can improve optimizations done by the compiler because it has more information than would be available in a traditional compiling process. This allows JIT compilation to make more performant choices as it compiles, which allows it to run more efficiently in certain cases.
```

Comment:
```suggestion
JIT compilation can improve optimizations done by the compiler because it has more information than what would be available in a traditional compiling process. This allows JIT compilation to make more performant choices as it compiles, which allows it to run more efficiently in certain cases.
```